jocd/src/br/org/certi/jocd/coresight/CortexM.java

```diff
@@ -1,9 +1,284 @@
+
+  // Debug Fault Status Register
+  public static final long DFSR = 0xE000ED30;
+  public static final int DFSR_EXTERNAL = (1 << 4);
+  public static final int DFSR_VCATCH = (1 << 3);
+  public static final int DFSR_DWTTRAP = (1 << 2);
+  public static final int DFSR_BKPT = (1 << 1);
+  public static final int DFSR_HALTED = (1 << 0);
+
+  // Debug Exception and Monitor Control Register
+  public static final long DEMCR = 0xE000EDFC;
+  // DWTENA in armv6 architecture reference manual
+  public static final int DEMCR_TRCENA = (1 << 24);
+  public static final int DEMCR_VC_HARDERR = (1 << 10);
+  public static final int DEMCR_VC_INTERR = (1 << 9);
+  public static final int DEMCR_VC_BUSERR = (1 << 8);
+  public static final int DEMCR_VC_STATERR = (1 << 7);
+  public static final int DEMCR_VC_CHKERR = (1 << 6);
+  public static final int DEMCR_VC_NOCPERR = (1 << 5);
+  public static final int DEMCR_VC_MMERR = (1 << 4);
+  public static final int DEMCR_VC_CORERESET = (1 << 0);
+
+  // CPUID Register
+  public static final long CPUID = 0xE000ED00;
+
+  // CPUID masks
+  public static final long CPUID_IMPLEMENTER_MASK = 0xff000000;
+  public static final int CPUID_IMPLEMENTER_POS = 24;
+  public static final long CPUID_VARIANT_MASK = 0x00f00000;
+  public static final int CPUID_VARIANT_POS = 20;
+  public static final long CPUID_ARCHITECTURE_MASK = 0x000f0000;
+  public static final int CPUID_ARCHITECTURE_POS = 16;
+  public static final long CPUID_PARTNO_MASK = 0x0000fff0;
+  public static final int CPUID_PARTNO_POS = 4;
+  public static final long CPUID_REVISION_MASK = 0x0000000f;
+  public static final int CPUID_REVISION_POS = 0;
+
+  public static final int CPUID_IMPLEMENTER_ARM = 0x41;
+  public static final int ARMv6M = 0xC;
+  public static final int ARMv7M = 0xF;
+
+  // Debug Core Register Selector Register
+  public static final long DCRSR = 0xE000EDF4;
+  public static final long DCRSR_REGWnR = (1 << 16);
+  public static final long DCRSR_REGSEL = 0x1F;
+
+  // Debug Halting Control and Status Register
+  public static final long DHCSR = 0xE000EDF0;
+  public static final int C_DEBUGEN = (1 << 0);
+  public static final int C_HALT = (1 << 1);
+  public static final int C_STEP = (1 << 2);
+  public static final int C_MASKINTS = (1 << 3);
+  public static final int C_SNAPSTALL = (1 << 5);
+  public static final int S_REGRDY = (1 << 16);
+  public static final int S_HALT = (1 << 17);
+  public static final int S_SLEEP = (1 << 18);
+  public static final int S_LOCKUP = (1 << 19);
+  public static final int S_RETIRE_ST = (1 << 24);
+  public static final int S_RESET_ST = (1 << 25);
+
+  // Debug Core Register Data Register
+  public static final long DCRDR = 0xE000EDF8;
+
+  // Coprocessor Access Control Register
+  public static final long CPACR = 0xE000ED88;
+  public static final int CPACR_CP10_CP11_MASK = (3 << 20) | (3 << 22);
+
+  public static final long NVIC_AIRCR = (0xE000ED0C);
```

Comment:
I don't think parentheses are needed for constants.

Reply:
Yes, it is not necessary. It is just a literal (and automated) from cortex_m.py.

Fixing...